docs/reference/extensions/dotnet-extensions.rst

```diff
@@ -0,0 +1,120 @@
+.. _reference-dotnet-extensions:
+
+.NET extensions
+===============
+
+The .NET extensions provide support for building and running .NET application snaps. They are versioned extensions,
+which means that you can specify which extension you want to use based on the .NET major version your application
+targets. The available extensions are:
+
+* ``dotnet8``: For applications targeting .NET 8 (``net8.0``)
+* ``dotnet9``: For applications targeting .NET 9 (``net9.0``)
+* ``dotnet10``: For applications targeting .NET 10 (``net10.0``)
+
+The .NET extensions are compatible with the core24 base.
+
+Included parts
+--------------
+
+The .NET extensions add the following parts to the project file:
+
+* ``prereqs``: A part that installs the prerequisite packages for running .NET applications. The installed packages are:
+
+   * ``libicu74``
+   * ``libssl3t64``
+   * ``libunwind8``
+   * ``liblttng-ust1t64``
+   * ``libbrotli1``
+
+* ``launcher``: A part that sets up the snap launcher to run the .NET application. It verifies whether the necessary
+  .NET runtime is available through a content interface connection and launches the application using the ``dotnet`` command.
+
+.. collapse:: Included parts
+
+    .. code-block:: yaml
+        :caption: snapcraft.yaml
+
+        launcher:
+            plugin: dump
+            source: /snap/snapcraft/x4/share/snapcraft/extensions/dotnet
+            override-build: |
+              mkdir -p $CRAFT_PART_INSTALL/bin/command-chain
+              cp launcher.sh $CRAFT_PART_INSTALL/bin/command-chain
+            stage:
+            - bin/command-chain/launcher.sh
+        prereqs:
+            plugin: nil
+            stage-packages:
+            - libicu74
+            - libunwind8
+            - libssl3t64
+            - liblttng-ust1t64
+            - libbrotli1
+
+Included plugs
+--------------
+
+The .NET extensions connect a snap-wide content plug to the .NET runtime content snap that matches the application's target
+.NET version.
+
+.. collapse:: Included snap-wide runtime plug for the dotnet8 extension
```

Comment:
This is a slight departure from what we normally do, but I'm warming up to it. Let's give it a try!

Reply:
You mean the collapsibles? I took inspiration from the other extensions' documentation, e.g., [Flutter](https://documentation.ubuntu.com/snapcraft/stable/reference/extensions/flutter-extension/) and [GNOME](https://documentation.ubuntu.com/snapcraft/stable/reference/extensions/gnome-extension/).